app/controllers/users_controller.rb

```diff
@@ -6,10 +6,17 @@ def show
 
   def edit
     @user = user
+    @user.avatar.attach(params[:avatar])
   end
 
   def update
     @user = user
+    # if @user.avatar.attached?
+    #   @user.avatar.purge
+    #   @user.avatar.attach(params[:avatar])
+    # else
+    #   @user.avatar.attach(params[:avatar])
+    # end
```

Comment:
Do you really need it?

Reply:
I followed  tips in the end of the article here https://medium.com/@anaharris/how-to-add-image-upload-functionality-to-your-rails-app-9f7fc3f3d042

Reply:
> Do you really need it?

No, removed it. It appeared that no need to add anything to controller at all.